app/models/host/managed.rb

```diff
@@ -29,11 +29,8 @@ class Host::Managed < Host::Base
   has_many :all_reports, :foreign_key => :host_id
 
   belongs_to :image
-  if Foreman::Plugin.installed?('foreman_host_reports')
-    has_many :host_statuses, -> { where("host_status.type != 'HostStatus::ConfigurationStatus' and host_status.type is not null") }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :delete_all
-  else
-    has_many :host_statuses, -> { where.not(type: nil) }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :destroy
-  end
+  has_many :host_statuses, -> { where.not(type: nil) }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :destroy
+  has_many :host_statuses_without_configuration, -> { where("host_status.type != 'HostStatus::ConfigurationStatus' and host_status.type is not null") }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host, :dependent => :destroy
```

Comment:
Only one of the associations should have the dependend: something, otherwise some of the records might be attempted to be destroyed twice.
```suggestion
  has_many :host_statuses_without_configuration, -> { where("host_status.type != 'HostStatus::ConfigurationStatus' and host_status.type is not null") }, :class_name => 'HostStatus::Status', :foreign_key => 'host_id', :inverse_of => :host
```

Reply:
btw, this would be probably better as a scope on the statuses themself as `scope :without_configuration`

Reply:
Good observation!

Reply:
Although it worked fine, I am removing.